### ISSUE TITLE
Github warning when using Issue reporter

### ISSUE THREAD

**Issue by rlabbe** [spyder-ide/spyder#11059]
### What steps will reproduce the problem?

I filed an issue using the Issue reporter. It asked me more my github credentials, which I filled in. Moments later I received the following email from github: 

----
You recently used a password to access an endpoint through the GitHub API using Python-urllib/3.7. We will deprecate basic authentication using password to this endpoint soon:

https://api.github.com/repositories/30892477/issues

We recommend using a personal access token (PAT) with the appropriate scope to access this endpoint instead. Visit https://github.com/settings/tokens for more information.

------

I submitted this report using Issue reporter, this time generating and using a PAT, and this worked. But googling suggests spyder should register itself as an OAuth application? https://developer.github.com/changes/2019-11-05-deprecated-passwords-and-authorizations-api/

In any case, it is a bit confusing to be allowed to log using a password and then get emails warning you about them. 

## Versions

* Spyder version: 4.0.0 
* Python version: 3.7.3
* Qt version: 5.9.6
* PyQt5 version: 5.9.2
* Operating System: Windows 7

### Dependencies

```
cloudpickle >=0.5.0          :  1.2.2 (OK)
pygments >=2.0               :  2.5.2 (OK)
qtconsole >=4.6.0            :  4.6.0 (OK)
nbconvert >=4.0              :  5.6.1 (OK)
sphinx >=0.6.6               :  2.2.2 (OK)
pylint >=0.25                :  2.4.4 (OK)
psutil >=0.3                 :  5.6.7 (OK)
qtawesome >=0.5.7            :  0.6.0 (OK)
qtpy >=1.5.0                 :  1.9.0 (OK)
pickleshare >=0.4            :  0.7.5 (OK)
zmq >=17                     :  18.1.0 (OK)
chardet >=2.0.0              :  3.0.4 (OK)
numpydoc >=0.6.0             :  0.9.1 (OK)
spyder_kernels >=1.8.1;<2.0.0:  1.8.1 (OK)
qdarkstyle >=2.7             :  2.7 (OK)
atomicwrites >=1.2.0         :  1.3.0 (OK)
diff_match_patch >=20181111  :  20181111 (OK)
intervaltree                 :  None (OK)
watchdog                     :  None (OK)
keyring                      :  None (OK)
pexpect >=4.4.0              :  4.7.0 (OK)
pympler                      :  None (OK)
sympy >=0.7.3                :  1.4 (OK)
cython >=0.21                :  0.29.14 (OK)
IPython >=4.0                :  7.10.1 (OK)
matplotlib >=2.0.0           :  3.1.1 (OK)
pandas >=0.13.1              :  0.25.3 (OK)
numpy >=1.7                  :  1.17.4 (OK)
scipy >=0.17.0               :  1.3.2 (OK)
pyls >=0.31.2;<0.32.0        :  0.31.2 (OK)
rtree >=0.8.3                :  0.8.3 (OK)
```

**Comment by ccordoba12** [comment]
> I submitted this report using Issue reporter, this time generating and using a PAT, and this worked.

It seems we need to disable password-based authentications and only request tokens. Thanks for letting us know about it.

> But googling suggests spyder should register itself as an OAuth application?

That only works for web apps. Non-web apps need to require personal tokens:

https://developer.github.com/apps/building-oauth-apps/authorizing-oauth-apps/#non-web-application-flow

**Comment by ccordoba12** [comment]
I decided to fix this for **4.0.1** because Github could remove password based authentications at any moment.
